### PR TITLE
Adding a KCor Map source

### LINKED ISSUE #1393: Add NCAR KCOR Data
The HAO has a coronagraph called KCOR which provides observations down to very close to the Sun. You can browse the available data here http://mlso.hao.ucar.edu/mlso_data_KCOR_2015.php. FITS files are provided.

### PR DESCRIPTION
**Pull request by wafels**
### Description

Supports KCor FITS data in SunPy.

Fixes #1393, and https://github.com/Helioviewer-Project/helioviewer.org/issues/239 .

### CODE REVIEW

File: sunpy/map/sources/mlso.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+from sunpy.map import GenericMap
+from sunpy.map.sources.source_type import source_stretch
+from sunpy.coordinates import get_sunearth_distance
+
+__all__ = ['KCorMap']
+
+
+class KCorMap(GenericMap):
+    """K-Cor Image Map.
```
Comment (nabobalis) [inline sunpy/map/sources/mlso.py]: Probably have a line break between the text and the triple quotes.

File: sunpy/map/sources/tests/test_kcor_source.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+"""Test cases for KCor Map subclass.
+This particular test file pertains to KCorMap.
+"""
```
Comment (nabobalis) [inline sunpy/map/sources/tests/test_kcor_source.py]: Also here, I think we can remove it.

File: sunpy/cm/cm.py [hunk 3]
```diff
@@ -63,6 +63,9 @@
 
 hmimag = ct.hmi_mag_color_table()
 
+kcor = deepcopy(mplcm.get_cmap("gist_gray"))
```
[the remark this answers is not in the document]
Comment (wafels) [reply sunpy/cm/cm.py]: The colormap needs its own name, so it really needs to be different.

File: sunpy/map/sources/mlso.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+    References
+    ----------
+    * `COSMO Mission Page <https://www2.hao.ucar.edu/cosmo>`_
+    * `KCOR Instrument Page <https://www2.hao.ucar.edu/mlso/instruments/mlso-kcor-coronagraph>`_
+    """
+
+    def __init__(self, data, header, **kwargs):
+
+        GenericMap.__init__(self, data, header, **kwargs)
```
Comment (Cadair) [inline sunpy/map/sources/mlso.py]: This should be like this now:

```suggestion
        super()(data, header, **kwargs)
```

File: sunpy/map/sources/mlso.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+        super().__init__(data, header, **kwargs)
+
+        # Fill in some missing info
+        self.meta['observatory'] = 'MLSO'
+        self.meta['detector'] = 'KCor'
+        self.meta['waveunit'] = 'nanometer'
+        # Since KCor is on Earth, no need to raise the warning in mapbase
+        self.meta['dsun_obs'] = (get_sunearth_distance(self.date)).to(u.m).value
+        self.meta['hgln_obs'] = 0.0
```
Comment (Cadair) [inline sunpy/map/sources/mlso.py]: Do we also need to set `hglt_obs`?

**Comment by wafels** [comment]
No, it's in there under one of its synonyms.

**Comment by wafels** [comment]
As in, the header contains the heliographic latitude under a recognizable
synonym.

**Comment by nabobalis** [comment]
Test fail is unrelated and something I will fix again.

**Comment by Cadair** [comment]
Thanks @wafels !

**Comment by nabobalis** [comment]
(Grumbles about the merge in master commit)

**Pull request merged**